ttnn/cpp/ttnn/operations/reduction/prod/prod.hpp

```diff
@@ -0,0 +1,56 @@
+// SPDX-FileCopyrightText: © 2024 Tenstorrent Inc.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#pragma once
+
+#include <optional>
+#include <functional>
+
+#include "ttnn/decorators.hpp"
+#include "ttnn/operations/core/core.hpp"
+
+namespace ttnn {
+
+namespace operations {
+
+namespace reduction {
+
+    struct ProdOp {
```

Comment:
why do we need this struct?

Reply:
It looks to me that it should be removed. Implementation should be used in execute methods themself in cpp

Reply:
@ayerofieiev-tt I think this was added to keep the _prod (or other _fns ) static or inline ?? some comment from Denys I saw with another issue. since we dont want the implementation in hpp files.
If it is ok to use the _prod from .cpp directly in Execute method then I can get rid of this ProdOp struct

Reply:
If you put the implementation in a header you typically want to make it inline.  But you can just use a static method forward declared in the header and implemented in the cpp.   The struct is unnecessary.

Reply:
Not sure if we even need those in a header at all